scripts/zkai_d128_proof_native_two_slice_compression_gate.py

```
@@ -0,0 +1,1116 @@
+    expect_keys(preimage, COMPRESSED_ARTIFACT_PREIMAGE_KEYS, "compressed artifact preimage")
+    public_inputs = require_object(preimage.get("proof_native_public_input_contract"), "compressed public inputs")
+    expect_keys(public_inputs, COMPRESSED_PUBLIC_INPUT_KEYS, "compressed public inputs")
+    for index, raw_entry in enumerate(require_list(preimage.get("compressed_transcript"), "compressed transcript")):
+        entry = require_object(raw_entry, f"compressed transcript entry {index}")
+        expect_keys(entry, COMPRESSED_TRANSCRIPT_ENTRY_KEYS, f"compressed transcript entry {index}")
+    source = copy.deepcopy(source) if source is not None else load_checked_accumulator()
+    expected_artifact = compressed_artifact(source)
+    expect_equal(artifact, expected_artifact, "compressed artifact")
+
+
+def verify_verifier_handle(handle: Any, artifact: dict[str, Any], source: dict[str, Any]) -> None:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `handle`

(ANN401)